source/elements/oneMKL/source/domains/dft/enums_and_config_params.rst

```diff
@@ -262,6 +265,16 @@ the :ref:`descriptor<onemkl_dft_descriptor>` class.
             :math:`M > 1`.
         -   | ``std::int64_t``
             | [0]
+      * -   :ref:`WORKSPACE_PLACEMENT<onemkl_dft_config_workspace_placement>`
+        -   Some FFT algorithm computation steps require a scratch space for permutations or other purposes. 
+            This parameter controls whether this scratch space is automatically allocated or provided by the user.
+        -   | :ref:`onemkl_dft_config_workspace_placement` (possible values are ``config_value::WORKSPACE_AUTOMATIC`` or ``config_value::WORKSPACE_EXTERNAL``).
+            | [``config_value::WORKSPACE_AUTOMATIC``]
+      * -   WORKSPACE_EXTERNAL_BYTES_REQUIRED
+        -   The required external workspace size in bytes when ``WORKSPACE_PLACEMENT`` is set to :ref:`config_value::WORKSPACE_EXTERNAL<onemkl_dft_config_value_workspace_automatic>`. 
+            A read-only value.
+        -   | ``std::int64_t``
```

Comment:
I think you're right in that it should be queryable - we state that a workspace can be set (but it may or may not do anything), even when `WORKSPACE_AUTOMATIC` is set. 

I don't think this should be used to query the internal memory requirements. For cases where external workspaces aren't supported (eg. MKL-CPU), we allow `WORKSPACE_EXTERNAL` to be set, but for the required space to always be zero. Even if we could obtain the workspace memory consumption of the descriptor in these cases, we wouldn't want to report it since the user might the allocate a non-zero external workspace that'd never be used.

Reply:
I've updated the doc string for `WORKSPACE_EXTERNAL_BYTES` in [e041739](https://github.com/oneapi-src/oneAPI-spec/pull/509/commits/e041739833d2e55d164382b70a0e48e6fe5530f1) to state that it is the minimum size allocation to use when calling `set_workspace`.

Reply:
Thank you!
Could you point me to the part about "we state that a workspace can be set (but it **may or may not** do anything), even when WORKSPACE_AUTOMATIC is set", please? Is that the documented behavior for some/many backends? I might have missed it but I find that a little odd.

If MKL-CPU is the only backend that does not support external workspace for now, wouldn't it be more suitable to throw an "unimplemented" exception if setting an external workspace with MKL-CPU and guarantee that the external workspace that was set is used, in other cases?

Reply:
I have *This function can be called on committed descriptors where the workspace placement is not config_value::WORKSPACE_EXTERNAL. The provided workspace may or may not be used in compute calls. However, the aforementioned restrictions will still apply.* in `set_workspace`. 

Regarding setting the workspace, when an external workspace was set pre-commit:
* In oneMKL and cuFFT, I believe setting the workspace causes the internally allocated workspace to be freed. 
* In rocFFT, the workspace is allocated and freed on each compute call by default. In oneMKL interface library, the workspace is set as an external workspace by the interface library to avoid this behaviour!
*  Many DFT libraries don't support external workspaces.
    * And this is what the *may not* bit of this is about.
    * I want it to be possible for the exact same code to work on all backends. 
        * The *may not* means that we can just ignore an external workspace when `set_workspace` is called on a backend that doesn't support it. This is in line with returning a zero-size required external workspace for these backends.

The rational for not stating *unsupported* is that:
* The user would require need a code-path to deal with external workspace not being supported.
* This extra codepath would presumably just not use an external workspace - we can do this internally anyway with very little effort.

Reply:
Thank you for the clarifications: that makes sense to me, personally!
Just requesting a confirmation that you meant "post-commit" and not "pre-commit" in that latest comment (thumbs-up if that was indeed a typo)?

Reply:
*Regarding setting the workspace, when an external workspace was set pre-commit:* is just wrong unfortunately. I should have said, *Regarding setting the workspace, when `WORKSPACE_EXTERNAL` was **not** set pre-commit (ie. `WORKSPACE_PLACEMENT` is `WORKSPACE_AUTOMATIC`):*.

Sorry for the confusion.